### PR TITLE
Add glnn

### PR DESCRIPTION
**Pull request by UNAOUN**
## Description
add glnn model

## Changes
1. add glnn to examples (train_teacher.py, train_student.py, train.conf.yaml, readme.md)
2. add get_train_val_test_split.py to utils, add get_train_val_test_split to utils/__init__.py
3. add get_train_val_test_split to docs/source/api/gammagl.utils.rst

### CODE REVIEW

File: examples/glnn/train_student.py [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+    teacher_logits = tlx.ops.convert_to_tensor(teacher_logits)
+
+    # for mindspore, it should be passed into node indices
+    if args.dataset in ['cora', 'pubmed', 'citeseer']:
+        train_idx = mask_to_index(graph.train_mask)
+        test_idx = mask_to_index(graph.test_mask)
+        val_idx = mask_to_index(graph.val_mask)
+        t_idx = tlx.concat([train_idx, test_idx, val_idx], axis=0)
+    elif args.dataset in ['computers', 'photo']:
+        train_mask, val_mask, test_mask = get_train_val_test_split(dataset, train_per_class=20, val_per_class=30)
```
Comment (dddg617) [inline examples/glnn/train_student.py:90]: Normally, you may get the mask through the `dataset.train_mask`. So, it would be better to modify the `gammagl.dataset.Amazon`.

File: examples/glnn/train_teacher.py [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+    edge_index = graph.edge_index
+    edge_weight = tlx.convert_to_tensor(calc_gcn_norm(edge_index, graph.num_nodes))
+
+    # for mindspore, it should be passed into node indices
+    if args.dataset in ['cora', 'pubmed', 'citeseer']:
+        train_idx = mask_to_index(graph.train_mask)
+        test_idx = mask_to_index(graph.test_mask)
+        val_idx = mask_to_index(graph.val_mask)
+    elif args.dataset in ['computers', 'photo']:
+        train_mask, val_mask, test_mask = get_train_val_test_split(dataset, train_per_class=20, val_per_class=30)
```
Comment (dddg617) [inline examples/glnn/train_teacher.py:80]: Ditto.

File: examples/glnn/train_student.py (hunk 1 above)
Comment (UNAOUN) [reply examples/glnn/train_student.py]: The requested modifications have been made to train_teacher.py, train_student.py, amazon.py, and get_split.py. 
Thank you for the reminder!
